shared/lib/gcs.py

```diff
@@ -26,13 +27,18 @@
 def download_file(bucket: str,
                   filename: str,
                   use_anonymous_client: bool = False) -> str:
+  local_embeddings_path = os.path.join(TEMP_DIR, filename)
+  if os.path.exists(local_embeddings_path):
+    logging.info("File already exists: %s, skipping GCS download.",
+                 local_embeddings_path)
+    return local_embeddings_path
+
```

Comment:
Looks like this is a general lib used by many callers?  To be sure, do we want to do this check just in the callsite?

nit:  Maybe rename "local_embeddings_path" to "local_file_path" ?